wrappers/binaries_test.go

```diff
@@ -0,0 +1,90 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2014-2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package wrappers_test
+
+import (
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+	"regexp"
+	"testing"
+
+	. "gopkg.in/check.v1"
+
+	"github.com/ubuntu-core/snappy/dirs"
+	"github.com/ubuntu-core/snappy/osutil"
+	"github.com/ubuntu-core/snappy/snap"
+	"github.com/ubuntu-core/snappy/snap/snaptest"
+	"github.com/ubuntu-core/snappy/wrappers"
+)
+
+func TestWrappers(t *testing.T) { TestingT(t) }
+
+type binariesTestSuite struct {
+	tempdir string
+}
+
+var _ = Suite(&binariesTestSuite{})
+
+func (s *binariesTestSuite) SetUpTest(c *C) {
+	s.tempdir = c.MkDir()
+	dirs.SetRootDir(s.tempdir)
+}
+
+func (s *binariesTestSuite) TearDownTest(c *C) {
+	dirs.SetRootDir("")
+}
+
+const packageHello = `name: hello-snap
+version: 1.10
+summary: hello
+description: Hello...
+apps:
+ hello:
+   command: bin/hello
+ svc1:
+  command: bin/hello
+  stop-command: bin/goodbye
+  post-stop-command: bin/missya
+  daemon: forking
+`
+
+func (s *binariesTestSuite) TestAddSnapBinariesAndRemove(c *C) {
+	info := snaptest.MockSnap(c, packageHello, &snap.SideInfo{Revision: 11})
+
+	err := wrappers.AddSnapBinaries(info)
+	c.Assert(err, IsNil)
+
+	wrapper := filepath.Join(s.tempdir, "/snap/bin/hello-snap.hello")
+
+	content, err := ioutil.ReadFile(wrapper)
+	c.Assert(err, IsNil)
+
+	needle := fmt.Sprintf(`
+ubuntu-core-launcher snap.hello-snap.hello snap.hello-snap.hello %s/snap/hello-snap/11/bin/hello "$@"
+`, s.tempdir)
```

Comment:
I think we want to test here for `/snap/hello-snap/11/bin/hello "$@"`, i.e. explicatively ensure that the GlobalRootDir is not used. Alternatively we need to find a new way for u-d-f